content/graph/lca.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Author: Thomas Meeks
+ * Date: 2024-10-08
+ * Source: own
+ * Description: Given a tree creates structure allowing for 
+ * O(log(n)) lca queries between arbitary nodes (root = 0)
+ * Time: O(Nlog(N)) Precomp, O(log(N)) query
+ * Status: Tested on https://cses.fi/problemset/task/1652
+ *              https://open.kattis.com/problems/tourists
+ */
+struct bl{
+    vector<int> dep; 
+    vector<vector<int>> jmp, g;
+    bl(vector<vector<int>> &g): 
+        dep(sz(g)),jmp(__lg(sz(g))+1,vector<int>(sz(g))),g(g){
+        dfs(0,0);
+        for(int lvl = 1; lvl < sz(jmp); lvl++)
+            for(int u = 0; u < sz(g); u++)
```

Comment:
You should check to see what everyone else thinks but I think u here should just be replaced with i since that's just the standard for loop variable. I don't think putting u here necessarily adds any clarity. Maybe also change the name of lvl as well bc its unnecessary characters but I definitely feel like that one makes more sense to keep.